skiko/src/nativeJsMain/cpp/FontStyleSet.cc

```diff
@@ -27,41 +27,24 @@ SKIKO_EXPORT KInt org_jetbrains_skia_FontStyleSet__1nGetStyle
 }
 
 
-SKIKO_EXPORT KInteropPointer org_jetbrains_skia_FontStyleSet__1nGetStyleName
-  (KNativePointer ptr, KInt index) {
-    TODO("implement org_jetbrains_skia_FontStyleSet__1nGetStyleName");
-}
-     
-#if 0 
 SKIKO_EXPORT KInteropPointer org_jetbrains_skia_FontStyleSet__1nGetStyleName
   (KNativePointer ptr, KInt index) {
     SkFontStyleSet* instance = reinterpret_cast<SkFontStyleSet*>((ptr));
     SkString style;
     instance->getStyle(index, nullptr, &style);
-    return javaString(env, style);
+    return new SkString(style);
```

Comment:
I'm not sure that it will be copy-elided, maybe use `std::move`?

Reply:
@VeselovAlex all I know about copy elision (but I know not that much however) is that std::move will do exactly the contrary - that is - it will eliminate any NRVO.

I'm not claiming though this code is better in that aspect )

Reply:
Not sure what you actually mean, we return the raw pointer, which by nature is "moved" :). What Shagen did is correct and intended way of returning Skia string pointers.

Reply:
@olonho I mean, when we call `new SkString(style)`, seems like the copy constructor is called, so the character buffer inside SkString will be copied. Example: https://godbolt.org/z/Ejh98dz6Y

Reply:
Ah, makes sense. Yes, it has COW `Rec` attribute.